htdocs/modulebuilder/template/core/modules/mymodule/doc/pdf_standard_myobject.modules.php

```diff
@@ -263,23 +263,25 @@ public function write_file($object, $outputlangs, $srctemplatepath = '', $hidede
 		*/
 
 		//if (count($realpatharray) == 0) $this->posxpicture=$this->posxtva;
+		$dir_output = getMultidirOutput($object, $object->module);
+		if (!empty($dir_output)) {
+			$dir_output .= '/' . $object->element;
 
-		if (getMultidirOutput($object)) {
 			$object->fetch_thirdparty();
 
 			$dir = null;
 			// Definition of $dir and $file
 			if ($object->specimen) {
-				$dir = getMultidirOutput($object);
+				$dir = $dir_output;
 				$file = $dir."/SPECIMEN.pdf";
 			} else {
 				$objectref = dol_sanitizeFileName($object->ref);
-				$dir = getMultidirOutput($object)."/".$objectref;
```

Comment:
Removing getMultidirOutput($object) is strange 
The method was introduced to replace dir_output so we can gey the correct dir depending on the entity.